FlashCards.ZubePierreBasali/Database/DataTools.cs

```diff
@@ -0,0 +1,240 @@
+﻿using FlashCards.FlashCardsManager.Models;
+using Spectre.Console;
+using System.Configuration;
+using Dapper;
+using System.Data;
+using System.Data.SqlClient;
+
+using FlashCards.StudySessions;
+
+namespace FlashCards.Database
+{
+    internal class DataTools
+    {
+        internal string ConnectionString { get; set;}
+
+        internal void Initialization()
+        {
+            try
+            {
+                string sqlCommandText;
+                Configuration config = ConfigurationManager.OpenExeConfiguration(ConfigurationUserLevel.None);
+                try
+                {
+                    if (config.AppSettings.Settings["connectionString"].Value == "")
+                    {
+                        string connectionString = @$"Data Source=localhost;Integrated Security=SSPI;Initial Catalog=;TrustServerCertificate=True;";
+                        config.AppSettings.Settings["connectionString"].Value = connectionString;
+                        config.Save();
+                        using (SqlConnection connection = new(connectionString))
+                        {
+                            if (connection.State == ConnectionState.Closed) connection.Open();
+                            sqlCommandText = $@"CREATE DATABASE FlashCardsDB";
+                        sqlCommandText = "nejwrnw";
+                            SqlCommand command = connection.CreateCommand();
+                            command.CommandText = sqlCommandText;
+                            command.ExecuteNonQuery();
+                            connection.Close();
+                        }
+                    }
+                }
+                catch(Exception ex) { throw ex; }
+
+                try
+                {
+                    ConnectionString = $"Data Source=localhost;Initial Catalog=FlashCardsDB;Integrated Security=True;TrustServerCertificate=True;";
+                    using (SqlConnection connection = new(ConnectionString))
+                    {
+                        connection.Open();
+                        SqlCommand command = connection.CreateCommand();
+                        sqlCommandText = @"IF OBJECT_ID(N'dbo.Stacks', N'U') IS NULL CREATE TABLE Stacks (Name nvarchar(20) PRIMARY KEY,NumberOfCards INTEGER,Id INTEGER);
+
+                                           IF OBJECT_ID(N'dbo.FlashCards', N'U') IS NULL CREATE TABLE FlashCards (Stack nvarchar(20),Id INTEGER, Question TEXT,Answer TEXT);
+                                           
+                                           IF NOT (EXISTS(SELECT * FROM INFORMATION_SCHEMA.TABLE_CONSTRAINTS WHERE CONSTRAINT_NAME = 'FK_FlashCards_Stacks')) ALTER TABLE [dbo].[FlashCards]
```

Comment:
🟠 **Inefficient**

❗ This needs refactoring! The Primary Keys should be IDs. The Foreign Keys should be on IDs.

💡 If you did this because of the unique stack name requirements. Then look into Unique Indexes. And/Or enforce with validation.